Compilation error in master branch using Clang 15.0.2 in arm64-apple-darwin21.6.0

**Description**
Compilation error in master branch using Clang 15.0.2 in arm64-apple-darwin21.6.0

**Scope**
Where in Kratos is this bug happening. 
E.g.
- Compilation
- KratosCore
- FooApplication
- Documentation

**To Reproduce**
Steps to reproduce the bug (if apply). Add any custom example files here if relevant.

configure.sh
```sh
#!/bin/bash
# Please do not modify this script

# You can use your interpreter of choice (bash, sh, zsh, ...)

# For any question please contact with us in:
#   - https://github.com/KratosMultiphysics/Kratos

# Optional parameters:
# You can find a list will all the compiation options in INSTALL.md or here:
#   - https://github.com/KratosMultiphysics/Kratos/wiki/Compilation-options

# Function to add apps
add_app () {
    export KRATOS_APPLICATIONS="${KRATOS_APPLICATIONS}$1;"
}

# Set compiler
export CC=/opt/homebrew/opt/llvm/bin/clang
export CXX=/opt/homebrew/opt/llvm/bin/clang++

# Set variables
export KRATOS_SOURCE="${KRATOS_SOURCE:-"$( cd "$(dirname "$0")" ; pwd -P )"/..}"
export KRATOS_BUILD="${KRATOS_SOURCE}/build"
export KRATOS_APP_DIR="${KRATOS_SOURCE}/applications"
export KRATOS_INSTALL_PYTHON_USING_LINKS=ON

# Set basic configuration
export KRATOS_BUILD_TYPE=${KRATOS_BUILD_TYPE:-"Release"}
export BOOST_ROOT=${BOOST_ROOT:-"/opt/homebrew/Cellar/boost/1.80.0"}
export PYTHON_EXECUTABLE=${PYTHON_EXECUTABLE:-"./venv/bin/python3"}

# Set applications to compile
export KRATOS_APPLICATIONS=
add_app ${KRATOS_APP_DIR}/LinearSolversApplication
add_app ${KRATOS_APP_DIR}/StructuralMechanicsApplication
add_app ${KRATOS_APP_DIR}/FluidDynamicsApplication

# Clean
clear
rm -rf "${KRATOS_BUILD}/${KRATOS_BUILD_TYPE}/cmake_install.cmake"
rm -rf "${KRATOS_BUILD}/${KRATOS_BUILD_TYPE}/CMakeCache.txt"
rm -rf "${KRATOS_BUILD}/${KRATOS_BUILD_TYPE}/CMakeFiles"

# Configure
cmake -H"${KRATOS_SOURCE}" -B"${KRATOS_BUILD}/${KRATOS_BUILD_TYPE}"    \
 -DCMAKE_CXX_FLAGS="${CMAKE_CXX_FLAGS} -std=c++17"                         \
 -DCMAKE_C_FLAGS="${CMAKE_C_FLAGS}"                                        \
 -DUSE_EIGEN_MKL=OFF                                                                                        \
 -DKRATOS_GENERATE_PYTHON_STUBS=OFF

# Buid
cmake --build "${KRATOS_BUILD}/${KRATOS_BUILD_TYPE}" --target install # -- -j$(sysctl -n hw.physicalcpu)
```

**OUTPUT**

```
-- The C compiler identification is Clang 15.0.2
-- The CXX compiler identification is Clang 15.0.2
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /opt/homebrew/opt/llvm/bin/clang - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /opt/homebrew/opt/llvm/bin/clang++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
CMake Deprecation Warning at CMakeLists.txt:2 (cmake_minimum_required):
  Compatibility with CMake < 2.8.12 will be removed from a future version of
  CMake.

  Update the VERSION argument <min> value or use a ...<max> suffix to tell
  CMake that the project does not need compatibility with older versions.


-- No KRATOS_SOURCE_DIR is defined, using: /Users/greatv/repos/Kratos
-- No KRATOS_BINARY_DIR is defined, using: /Users/greatv/repos/Kratos/build/Release
-- No INSTALL_RUNKRATOS is defined, setting to ON
-- No CMAKE_INSTALL_MESSAGE is defined. Setting to NEVER
-- Found Git: /usr/bin/git (found version "2.37.0 (Apple Git-136)") 
-- CMAKE_SYSTEM_NAME = Darwin
-- CMAKE_CXX_COMPILER_ID = Clang
-- CMAKE_CXX_FLAGS =  -std=c++17
-- CMAKE_C_FLAGS = 
-- Found PythonInterp: ./venv/bin/python3 (found version "3.10.7") 
-- Found PythonLibs: /opt/homebrew/opt/python@3.10/Frameworks/Python.framework/Versions/3.10/lib/libpython3.10.dylib
-- LTO is Disabled
AR VERSION: /usr/bin/ar
-- Python version used for the interface will be 3.10
-- Standard install dir /Users/greatv/repos/Kratos/bin/Release
"KRATOS_SHARED_MEMORY_PARALLELIZATION" not defined, defaulting to "OpenMP"
-- Found OpenMP_C: -fopenmp=libomp (found version "5.0") 
-- Found OpenMP_CXX: -fopenmp=libomp (found version "5.0") 
-- Found OpenMP: TRUE (found version "5.0")  
Using OpenMP for shared memory parallelization
-- Found Boost: /opt/homebrew/Cellar/boost/1.80.0/lib/cmake/Boost-1.80.0/BoostConfig.cmake (found version "1.80.0")  
-- Boost Include: /opt/homebrew/Cellar/boost/1.80.0/include
-- Boost Linkdir: 
-- Found ZLIB: /Library/Developer/CommandLineTools/SDKs/MacOSX12.3.sdk/usr/lib/libz.tbd (found version "1.2.11") 
-- AMatrix not found. Please define the AMATRIX_DIR pointing to the AMatrix root directory.
 
CMAKE_BUILD_TYPE ........ Release
 
WARNING: YOUR GIDPOST IS BEING COMPILED  WITHOUT DEBUG (ALL ASSERTS WILL BE AVOIDED)
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD - Success
-- Found Threads: TRUE  
-- Configuring applications (ENV):
Adding application '/Users/greatv/repos/Kratos/scripts/../applications/LinearSolversApplication'
**** configuring KratosLinearSolversApplication ****
Adding application '/Users/greatv/repos/Kratos/scripts/../applications/StructuralMechanicsApplication'
**** configuring KratosStructuralMechanicsApplication ****
Adding application '/Users/greatv/repos/Kratos/scripts/../applications/FluidDynamicsApplication'
**** configuring KratosFluidDynamicsApplication ****

***********************************************************************

        LIST OF APPLICATIONS THAT ARE GOING TO BE COMPILED:

        LinearSolversApplication
        StructuralMechanicsApplication
        FluidDynamicsApplication

        APPLICATIONS ADDED TO COMPILATION THROUGH DEPENDENCIES:


***********************************************************************

List of upcoming API-breaking and behavior changes:
         Geometry old projection methods deprecation: https://github.com/KratosMultiphysics/Kratos/pull/9024
         Geometry old closest point methods deprecation: https://github.com/KratosMultiphysics/Kratos/pull/9243
         Model 'CreateModelPart' behavior change (no error thrown if model part exists): https://github.com/KratosMultiphysics/Kratos/pull/9598
         Make GetIntegrationMethod method of Condition const: https://github.com/KratosMultiphysics/Kratos/pull/9769
         Behavior change of ModelPart.GetProperties (MeshIndex removed): https://github.com/KratosMultiphysics/Kratos/pull/9774
-- tinyexpr : /Users/greatv/repos/Kratos/external_libraries/tinyexpr/tinyexpr/
python libraries to be installed are: /opt/homebrew/Cellar/python@3.10/3.10.7/Frameworks/Python.framework/Versions/3.10/Python
-- Configuring done
CMake Warning (dev):
  Policy CMP0042 is not set: MACOSX_RPATH is enabled by default.  Run "cmake
  --help-policy CMP0042" for policy details.  Use the cmake_policy command to
  set the policy and suppress this warning.

  MACOSX_RPATH is not specified for the following targets:

   KratosCore
   KratosFluidDynamicsCore
   KratosLinearSolversCore
   KratosStructuralMechanicsCore

This warning is for project developers.  Use -Wno-dev to suppress it.

-- Generating done
-- Build files have been written to: /Users/greatv/repos/Kratos/build/Release
Scanning dependencies of target gidpost
[  1%] Built target gidpost
Consolidate compiler generated dependencies of target KratosVersion
[  1%] Built target KratosVersion
Scanning dependencies of target tinyexpr
[  2%] Built target tinyexpr
Scanning dependencies of target KratosCore
Consolidate compiler generated dependencies of target KratosCore
[ 56%] Built target KratosCore
Consolidate compiler generated dependencies of target Kratos
[ 63%] Built target Kratos
Consolidate compiler generated dependencies of target KratosLinearSolversCore
[ 64%] Built target KratosLinearSolversCore
Consolidate compiler generated dependencies of target KratosLinearSolversApplication
[ 64%] Built target KratosLinearSolversApplication
Consolidate compiler generated dependencies of target KratosStructuralMechanicsCore
[ 82%] Built target KratosStructuralMechanicsCore
Consolidate compiler generated dependencies of target KratosStructuralMechanicsApplication
[ 83%] Built target KratosStructuralMechanicsApplication
Consolidate compiler generated dependencies of target KratosFluidDynamicsCore
[ 83%] Building CXX object applications/FluidDynamicsApplication/CMakeFiles/KratosFluidDynamicsCore.dir/custom_processes/distance_smoothing_process.cpp.o
In file included from /Users/greatv/repos/Kratos/applications/FluidDynamicsApplication/custom_processes/distance_smoothing_process.cpp:16:
In file included from /Users/greatv/repos/Kratos/applications/FluidDynamicsApplication/custom_processes/distance_smoothing_process.h:23:
In file included from /Users/greatv/repos/Kratos/kratos/processes/find_global_nodal_neighbours_process.h:28:
/Users/greatv/repos/Kratos/kratos/utilities/pointer_communicator.h:67:27: warning: 'result_of<(lambda at /Users/greatv/repos/Kratos/applications/FluidDynamicsApplication/custom_processes/distance_smoothing_process.h:203:13) (Kratos::GlobalPointer<Kratos::Node<3>> &)>' is deprecated [-Wdeprecated-declarations]
    typedef typename std::result_of< TFunctorType(GlobalPointer<TPointerDataType>&)>::type TSendType;
                          ^
/Users/greatv/repos/Kratos/applications/FluidDynamicsApplication/custom_processes/distance_smoothing_process.h:202:44: note: in instantiation of template class 'Kratos::ResultsProxy<Kratos::Node<3>, (lambda at /Users/greatv/repos/Kratos/applications/FluidDynamicsApplication/custom_processes/distance_smoothing_process.h:203:13)>' requested here
        auto combined_proxy = pointer_comm.Apply(
                                           ^
/Users/greatv/repos/Kratos/applications/FluidDynamicsApplication/custom_processes/distance_smoothing_process.cpp:25:16: note: in instantiation of member function 'Kratos::DistanceSmoothingProcess<2, Kratos::UblasSpace<double, boost::numeric::ublas::compressed_matrix<double, boost::numeric::ublas::basic_row_major<>, 0>, boost::numeric::ublas::vector<double>>, Kratos::UblasSpace<double, boost::numeric::ublas::matrix<double>, boost::numeric::ublas::vector<double>>, Kratos::LinearSolver<Kratos::UblasSpace<double, boost::numeric::ublas::compressed_matrix<double, boost::numeric::ublas::basic_row_major<>, 0>, boost::numeric::ublas::vector<double>>, Kratos::UblasSpace<double, boost::numeric::ublas::matrix<double>, boost::numeric::ublas::vector<double>>>>::Execute' requested here
template class DistanceSmoothingProcess<2,SparseSpaceType,LocalSpaceType,LinearSolverType>;
               ^
/opt/homebrew/opt/llvm/bin/../include/c++/v1/type_traits:578:34: note: 'result_of<(lambda at /Users/greatv/repos/Kratos/applications/FluidDynamicsApplication/custom_processes/distance_smoothing_process.h:203:13) (Kratos::GlobalPointer<Kratos::Node<3>> &)>' has been explicitly marked deprecated here
template <class _Callable> class _LIBCPP_DEPRECATED_IN_CXX17 result_of;
                                 ^
/opt/homebrew/opt/llvm/bin/../include/c++/v1/__config:800:41: note: expanded from macro '_LIBCPP_DEPRECATED_IN_CXX17'
#    define _LIBCPP_DEPRECATED_IN_CXX17 _LIBCPP_DEPRECATED
                                        ^
/opt/homebrew/opt/llvm/bin/../include/c++/v1/__config:773:49: note: expanded from macro '_LIBCPP_DEPRECATED'
#      define _LIBCPP_DEPRECATED __attribute__((deprecated))
                                                ^
In file included from /Users/greatv/repos/Kratos/applications/FluidDynamicsApplication/custom_processes/distance_smoothing_process.cpp:16:
In file included from /Users/greatv/repos/Kratos/applications/FluidDynamicsApplication/custom_processes/distance_smoothing_process.h:23:
In file included from /Users/greatv/repos/Kratos/kratos/processes/find_global_nodal_neighbours_process.h:28:
/Users/greatv/repos/Kratos/kratos/utilities/pointer_communicator.h:67:27: warning: 'result_of<(lambda at /Users/greatv/repos/Kratos/applications/FluidDynamicsApplication/custom_processes/distance_smoothing_process.h:210:13) (Kratos::GlobalPointer<Kratos::Node<3>> &)>' is deprecated [-Wdeprecated-declarations]
    typedef typename std::result_of< TFunctorType(GlobalPointer<TPointerDataType>&)>::type TSendType;
                          ^
/Users/greatv/repos/Kratos/applications/FluidDynamicsApplication/custom_processes/distance_smoothing_process.h:209:43: note: in instantiation of template class 'Kratos::ResultsProxy<Kratos::Node<3>, (lambda at /Users/greatv/repos/Kratos/applications/FluidDynamicsApplication/custom_processes/distance_smoothing_process.h:210:13)>' requested here
        auto contact_proxy = pointer_comm.Apply(
                                          ^
/Users/greatv/repos/Kratos/applications/FluidDynamicsApplication/custom_processes/distance_smoothing_process.cpp:25:16: note: in instantiation of member function 'Kratos::DistanceSmoothingProcess<2, Kratos::UblasSpace<double, boost::numeric::ublas::compressed_matrix<double, boost::numeric::ublas::basic_row_major<>, 0>, boost::numeric::ublas::vector<double>>, Kratos::UblasSpace<double, boost::numeric::ublas::matrix<double>, boost::numeric::ublas::vector<double>>, Kratos::LinearSolver<Kratos::UblasSpace<double, boost::numeric::ublas::compressed_matrix<double, boost::numeric::ublas::basic_row_major<>, 0>, boost::numeric::ublas::vector<double>>, Kratos::UblasSpace<double, boost::numeric::ublas::matrix<double>, boost::numeric::ublas::vector<double>>>>::Execute' requested here
template class DistanceSmoothingProcess<2,SparseSpaceType,LocalSpaceType,LinearSolverType>;
               ^
/opt/homebrew/opt/llvm/bin/../include/c++/v1/type_traits:578:34: note: 'result_of<(lambda at /Users/greatv/repos/Kratos/applications/FluidDynamicsApplication/custom_processes/distance_smoothing_process.h:210:13) (Kratos::GlobalPointer<Kratos::Node<3>> &)>' has been explicitly marked deprecated here
template <class _Callable> class _LIBCPP_DEPRECATED_IN_CXX17 result_of;
                                 ^
/opt/homebrew/opt/llvm/bin/../include/c++/v1/__config:800:41: note: expanded from macro '_LIBCPP_DEPRECATED_IN_CXX17'
#    define _LIBCPP_DEPRECATED_IN_CXX17 _LIBCPP_DEPRECATED
                                        ^
/opt/homebrew/opt/llvm/bin/../include/c++/v1/__config:773:49: note: expanded from macro '_LIBCPP_DEPRECATED'
#      define _LIBCPP_DEPRECATED __attribute__((deprecated))
                                                ^
In file included from /Users/greatv/repos/Kratos/applications/FluidDynamicsApplication/custom_processes/distance_smoothing_process.cpp:16:
In file included from /Users/greatv/repos/Kratos/applications/FluidDynamicsApplication/custom_processes/distance_smoothing_process.h:23:
In file included from /Users/greatv/repos/Kratos/kratos/processes/find_global_nodal_neighbours_process.h:28:
/Users/greatv/repos/Kratos/kratos/utilities/pointer_communicator.h:67:27: warning: 'result_of<(lambda at /Users/greatv/repos/Kratos/applications/FluidDynamicsApplication/custom_processes/distance_smoothing_process.h:203:13) (Kratos::GlobalPointer<Kratos::Node<3>> &)>' is deprecated [-Wdeprecated-declarations]
    typedef typename std::result_of< TFunctorType(GlobalPointer<TPointerDataType>&)>::type TSendType;
                          ^
/Users/greatv/repos/Kratos/applications/FluidDynamicsApplication/custom_processes/distance_smoothing_process.h:202:44: note: in instantiation of template class 'Kratos::ResultsProxy<Kratos::Node<3>, (lambda at /Users/greatv/repos/Kratos/applications/FluidDynamicsApplication/custom_processes/distance_smoothing_process.h:203:13)>' requested here
        auto combined_proxy = pointer_comm.Apply(
                                           ^
/Users/greatv/repos/Kratos/applications/FluidDynamicsApplication/custom_processes/distance_smoothing_process.cpp:26:16: note: in instantiation of member function 'Kratos::DistanceSmoothingProcess<3, Kratos::UblasSpace<double, boost::numeric::ublas::compressed_matrix<double, boost::numeric::ublas::basic_row_major<>, 0>, boost::numeric::ublas::vector<double>>, Kratos::UblasSpace<double, boost::numeric::ublas::matrix<double>, boost::numeric::ublas::vector<double>>, Kratos::LinearSolver<Kratos::UblasSpace<double, boost::numeric::ublas::compressed_matrix<double, boost::numeric::ublas::basic_row_major<>, 0>, boost::numeric::ublas::vector<double>>, Kratos::UblasSpace<double, boost::numeric::ublas::matrix<double>, boost::numeric::ublas::vector<double>>>>::Execute' requested here
template class DistanceSmoothingProcess<3,SparseSpaceType,LocalSpaceType,LinearSolverType>;
               ^
/opt/homebrew/opt/llvm/bin/../include/c++/v1/type_traits:578:34: note: 'result_of<(lambda at /Users/greatv/repos/Kratos/applications/FluidDynamicsApplication/custom_processes/distance_smoothing_process.h:203:13) (Kratos::GlobalPointer<Kratos::Node<3>> &)>' has been explicitly marked deprecated here
template <class _Callable> class _LIBCPP_DEPRECATED_IN_CXX17 result_of;
                                 ^
/opt/homebrew/opt/llvm/bin/../include/c++/v1/__config:800:41: note: expanded from macro '_LIBCPP_DEPRECATED_IN_CXX17'
#    define _LIBCPP_DEPRECATED_IN_CXX17 _LIBCPP_DEPRECATED
                                        ^
/opt/homebrew/opt/llvm/bin/../include/c++/v1/__config:773:49: note: expanded from macro '_LIBCPP_DEPRECATED'
#      define _LIBCPP_DEPRECATED __attribute__((deprecated))
                                                ^
In file included from /Users/greatv/repos/Kratos/applications/FluidDynamicsApplication/custom_processes/distance_smoothing_process.cpp:16:
In file included from /Users/greatv/repos/Kratos/applications/FluidDynamicsApplication/custom_processes/distance_smoothing_process.h:23:
In file included from /Users/greatv/repos/Kratos/kratos/processes/find_global_nodal_neighbours_process.h:28:
/Users/greatv/repos/Kratos/kratos/utilities/pointer_communicator.h:67:27: warning: 'result_of<(lambda at /Users/greatv/repos/Kratos/applications/FluidDynamicsApplication/custom_processes/distance_smoothing_process.h:210:13) (Kratos::GlobalPointer<Kratos::Node<3>> &)>' is deprecated [-Wdeprecated-declarations]
    typedef typename std::result_of< TFunctorType(GlobalPointer<TPointerDataType>&)>::type TSendType;
                          ^
/Users/greatv/repos/Kratos/applications/FluidDynamicsApplication/custom_processes/distance_smoothing_process.h:209:43: note: in instantiation of template class 'Kratos::ResultsProxy<Kratos::Node<3>, (lambda at /Users/greatv/repos/Kratos/applications/FluidDynamicsApplication/custom_processes/distance_smoothing_process.h:210:13)>' requested here
        auto contact_proxy = pointer_comm.Apply(
                                          ^
/Users/greatv/repos/Kratos/applications/FluidDynamicsApplication/custom_processes/distance_smoothing_process.cpp:26:16: note: in instantiation of member function 'Kratos::DistanceSmoothingProcess<3, Kratos::UblasSpace<double, boost::numeric::ublas::compressed_matrix<double, boost::numeric::ublas::basic_row_major<>, 0>, boost::numeric::ublas::vector<double>>, Kratos::UblasSpace<double, boost::numeric::ublas::matrix<double>, boost::numeric::ublas::vector<double>>, Kratos::LinearSolver<Kratos::UblasSpace<double, boost::numeric::ublas::compressed_matrix<double, boost::numeric::ublas::basic_row_major<>, 0>, boost::numeric::ublas::vector<double>>, Kratos::UblasSpace<double, boost::numeric::ublas::matrix<double>, boost::numeric::ublas::vector<double>>>>::Execute' requested here
template class DistanceSmoothingProcess<3,SparseSpaceType,LocalSpaceType,LinearSolverType>;
               ^
/opt/homebrew/opt/llvm/bin/../include/c++/v1/type_traits:578:34: note: 'result_of<(lambda at /Users/greatv/repos/Kratos/applications/FluidDynamicsApplication/custom_processes/distance_smoothing_process.h:210:13) (Kratos::GlobalPointer<Kratos::Node<3>> &)>' has been explicitly marked deprecated here
template <class _Callable> class _LIBCPP_DEPRECATED_IN_CXX17 result_of;
                                 ^
/opt/homebrew/opt/llvm/bin/../include/c++/v1/__config:800:41: note: expanded from macro '_LIBCPP_DEPRECATED_IN_CXX17'
#    define _LIBCPP_DEPRECATED_IN_CXX17 _LIBCPP_DEPRECATED
                                        ^
/opt/homebrew/opt/llvm/bin/../include/c++/v1/__config:773:49: note: expanded from macro '_LIBCPP_DEPRECATED'
#      define _LIBCPP_DEPRECATED __attribute__((deprecated))
                                                ^
In file included from /Users/greatv/repos/Kratos/applications/FluidDynamicsApplication/custom_processes/distance_smoothing_process.cpp:16:
In file included from /Users/greatv/repos/Kratos/applications/FluidDynamicsApplication/custom_processes/distance_smoothing_process.h:21:
In file included from /Users/greatv/repos/Kratos/kratos/processes/process.h:25:
In file included from /Users/greatv/repos/Kratos/kratos/includes/kratos_flags.h:27:
In file included from /Users/greatv/repos/Kratos/kratos/includes/kratos_components.h:27:
In file included from /Users/greatv/repos/Kratos/kratos/includes/data_communicator.h:27:
In file included from /Users/greatv/repos/Kratos/kratos/includes/mpi_serializer.h:26:
In file included from /Users/greatv/repos/Kratos/kratos/includes/stream_serializer.h:26:
/Users/greatv/repos/Kratos/kratos/includes/serializer.h:628:17: error: no member named 'save' in 'std::__bit_const_reference<std::vector<bool>>'
        rObject.save(*this);
        ~~~~~~~ ^
/Users/greatv/repos/Kratos/kratos/includes/serializer.h:569:13: note: in instantiation of function template specialization 'Kratos::Serializer::save<std::__bit_const_reference<std::vector<bool>>>' requested here
            save("E", rObject[i]);
            ^
/Users/greatv/repos/Kratos/kratos/includes/data_communicator.h:890:29: note: in instantiation of function template specialization 'Kratos::Serializer::save<bool>' requested here
            send_serializer.save("data", rSendObject);
                            ^
/Users/greatv/repos/Kratos/kratos/includes/data_communicator.h:531:22: note: in instantiation of function template specialization 'Kratos::DataCommunicator::SendRecvImpl<std::vector<bool>>' requested here
        return this->SendRecvImpl(rSendObject, SendDestination, 0, RecvSource, 0);
                     ^
/Users/greatv/repos/Kratos/kratos/utilities/pointer_communicator.h:238:55: note: in instantiation of function template specialization 'Kratos::DataCommunicator::SendRecv<std::vector<bool>>' requested here
                auto remote_data = mrDataCommunicator.SendRecv(locally_gathered_data, color, color );
                                                      ^
/Users/greatv/repos/Kratos/kratos/utilities/pointer_communicator.h:214:13: note: in instantiation of function template specialization 'Kratos::GlobalPointerCommunicator<Kratos::Node<3>>::Update<(lambda at /Users/greatv/repos/Kratos/applications/FluidDynamicsApplication/custom_processes/distance_smoothing_process.h:210:13)>' requested here
            Update(UserFunctor, non_local_data);
            ^
/Users/greatv/repos/Kratos/applications/FluidDynamicsApplication/custom_processes/distance_smoothing_process.h:209:43: note: in instantiation of function template specialization 'Kratos::GlobalPointerCommunicator<Kratos::Node<3>>::Apply<(lambda at /Users/greatv/repos/Kratos/applications/FluidDynamicsApplication/custom_processes/distance_smoothing_process.h:210:13)>' requested here
        auto contact_proxy = pointer_comm.Apply(
                                          ^
4 warnings and 1 error generated.
make[2]: *** [applications/FluidDynamicsApplication/CMakeFiles/KratosFluidDynamicsCore.dir/custom_processes/distance_smoothing_process.cpp.o] Error 1
make[1]: *** [applications/FluidDynamicsApplication/CMakeFiles/KratosFluidDynamicsCore.dir/all] Error 2
make: *** [all] Error 2
```
**Expected behavior**
A clear and concise description of what should be the expected behaviour.

**Environment**
Please specify your operating system, kratos branch, python version or any other library versions that you consider relevant.
E.g.:
- OS: MacOS 12.6
- Branch: master
- Python 3.10

> > This is the Homebrew Clang or the Mac OS clang?
> 
> Okay, I see it is homebrew. With gcc do you have any issue?
> 
> Maybe we should test with Clang 15 (the latest) and see if the problem is the version.
> 
> PS: With homebrew you can install older versions of compilers, you can check with an older one. Unfortunately, I don't have a M1 machine.

With homebrew gcc
```bash
#!/bin/bash
# Please do not modify this script

# You can use your interpreter of choice (bash, sh, zsh, ...)

# For any question please contact with us in:
#   - https://github.com/KratosMultiphysics/Kratos

# Optional parameters:
# You can find a list will all the compiation options in INSTALL.md or here:
#   - https://github.com/KratosMultiphysics/Kratos/wiki/Compilation-options

# Function to add apps
add_app () {
    export KRATOS_APPLICATIONS="${KRATOS_APPLICATIONS}$1;"
}

# Set compiler
# export CC=/opt/homebrew/opt/llvm/bin/clang
# export CXX=/opt/homebrew/opt/llvm/bin/clang++
export CC=/opt/homebrew/bin/gcc-12
export CXX=/opt/homebrew/bin/g++-12

# Set variables
export KRATOS_SOURCE="${KRATOS_SOURCE:-"$( cd "$(dirname "$0")" ; pwd -P )"/..}"
export KRATOS_BUILD="${KRATOS_SOURCE}/build"
export KRATOS_APP_DIR="${KRATOS_SOURCE}/applications"
export KRATOS_INSTALL_PYTHON_USING_LINKS=ON

# Set basic configuration
export KRATOS_BUILD_TYPE=${KRATOS_BUILD_TYPE:-"Release"}
export BOOST_ROOT=${BOOST_ROOT:-"/opt/homebrew/Cellar/boost/1.80.0"}
export PYTHON_EXECUTABLE=${PYTHON_EXECUTABLE:-"./venv/bin/python3"}

# Set applications to compile
export KRATOS_APPLICATIONS=
add_app ${KRATOS_APP_DIR}/LinearSolversApplication
add_app ${KRATOS_APP_DIR}/StructuralMechanicsApplication
add_app ${KRATOS_APP_DIR}/FluidDynamicsApplication

# Clean
clear
rm -rf "${KRATOS_BUILD}/${KRATOS_BUILD_TYPE}/cmake_install.cmake"
rm -rf "${KRATOS_BUILD}/${KRATOS_BUILD_TYPE}/CMakeCache.txt"
rm -rf "${KRATOS_BUILD}/${KRATOS_BUILD_TYPE}/CMakeFiles"

# Configure
cmake -H"${KRATOS_SOURCE}" -B"${KRATOS_BUILD}/${KRATOS_BUILD_TYPE}"    \
 -DCMAKE_CXX_FLAGS="${CMAKE_CXX_FLAGS} -std=c++17"                         \
 -DCMAKE_C_FLAGS="${CMAKE_C_FLAGS}"                                        \
 -DUSE_EIGEN_MKL=OFF                                                                                        \
 -DKRATOS_GENERATE_PYTHON_STUBS=OFF

# Buid
cmake --build "${KRATOS_BUILD}/${KRATOS_BUILD_TYPE}" --target install # -- -j$(sysctl -n hw.physicalcpu)

```

OpenMP could not be found!

```
-- The C compiler identification is GNU 12.2.0
-- The CXX compiler identification is GNU 12.2.0
-- Checking whether C compiler has -isysroot
-- Checking whether C compiler has -isysroot - yes
-- Checking whether C compiler supports OSX deployment target flag
-- Checking whether C compiler supports OSX deployment target flag - yes
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /opt/homebrew/bin/gcc-12 - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Checking whether CXX compiler has -isysroot
-- Checking whether CXX compiler has -isysroot - yes
-- Checking whether CXX compiler supports OSX deployment target flag
-- Checking whether CXX compiler supports OSX deployment target flag - yes
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /opt/homebrew/bin/g++-12 - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
CMake Deprecation Warning at CMakeLists.txt:2 (cmake_minimum_required):
  Compatibility with CMake < 2.8.12 will be removed from a future version of
  CMake.

  Update the VERSION argument <min> value or use a ...<max> suffix to tell
  CMake that the project does not need compatibility with older versions.


-- No KRATOS_SOURCE_DIR is defined, using: /Users/wangxin/repos/Kratos
-- No KRATOS_BINARY_DIR is defined, using: /Users/wangxin/repos/Kratos/build/Release
-- No INSTALL_RUNKRATOS is defined, setting to ON
-- No CMAKE_INSTALL_MESSAGE is defined. Setting to NEVER
-- Found Git: /usr/bin/git (found version "2.37.0 (Apple Git-136)") 
-- Additional default options were set for GCC
-- CMAKE_SYSTEM_NAME = Darwin
-- CMAKE_CXX_COMPILER_ID = GNU
-- CMAKE_CXX_FLAGS =  -std=c++17 -funroll-loops -Wall -Wl,--no-as-needed -ldl -Wsuggest-override
-- CMAKE_C_FLAGS =  -funroll-loops -Wall -Wl,--no-as-needed -ldl
-- Found PythonInterp: ./venv/bin/python3 (found version "3.10.7") 
-- Found PythonLibs: /opt/homebrew/opt/python@3.10/Frameworks/Python.framework/Versions/3.10/lib/libpython3.10.dylib
-- LTO is Disabled
AR VERSION: /usr/bin/ar
-- Python version used for the interface will be 3.10
-- Standard install dir /Users/wangxin/repos/Kratos/bin/Release
"KRATOS_SHARED_MEMORY_PARALLELIZATION" not defined, defaulting to "OpenMP"
-- Could NOT find OpenMP_C (missing: OpenMP_C_FLAGS OpenMP_C_LIB_NAMES) 
-- Could NOT find OpenMP_CXX (missing: OpenMP_CXX_FLAGS OpenMP_CXX_LIB_NAMES) 
-- Could NOT find OpenMP (missing: OpenMP_C_FOUND OpenMP_CXX_FOUND) 
CMake Error at CMakeLists.txt:335 (message):
  OpenMP could not be found!


-- Configuring incomplete, errors occurred!
See also "/Users/wangxin/repos/Kratos/build/Release/CMakeFiles/CMakeOutput.log".
See also "/Users/wangxin/repos/Kratos/build/Release/CMakeFiles/CMakeError.log".
make: Makefile: No such file or directory
make: *** No rule to make target `Makefile'.  Stop.
```

@matekelemen this works 👍 

**compile output:**

```
In file included from /Users/wangxin/repos/Kratos/applications/FluidDynamicsApplication/custom_python/add_custom_processes_to_python.cpp:35:
In file included from /Users/wangxin/repos/Kratos/applications/FluidDynamicsApplication/custom_processes/distance_smoothing_process.h:23:
In file included from /Users/wangxin/repos/Kratos/kratos/processes/find_global_nodal_neighbours_process.h:28:
/Users/wangxin/repos/Kratos/kratos/utilities/pointer_communicator.h:67:27: warning: 'result_of<(lambda at /Users/wangxin/repos/Kratos/applications/FluidDynamicsApplication/custom_processes/distance_smoothing_process.h:203:13) (Kratos::GlobalPointer<Kratos::Node<3>> &)>' is deprecated [-Wdeprecated-declarations]
    typedef typename std::result_of< TFunctorType(GlobalPointer<TPointerDataType>&)>::type TSendType;
                          ^
/Users/wangxin/repos/Kratos/applications/FluidDynamicsApplication/custom_processes/distance_smoothing_process.h:202:44: note: in instantiation of template class 'Kratos::ResultsProxy<Kratos::Node<3>, (lambda at /Users/wangxin/repos/Kratos/applications/FluidDynamicsApplication/custom_processes/distance_smoothing_process.h:203:13)>' requested here
        auto combined_proxy = pointer_comm.Apply(
                                           ^
/Users/wangxin/repos/Kratos/applications/FluidDynamicsApplication/custom_processes/distance_smoothing_process.h:152:5: note: in instantiation of member function 'Kratos::DistanceSmoothingProcess<3, Kratos::UblasSpace<double, boost::numeric::ublas::compressed_matrix<double, boost::numeric::ublas::basic_row_major<>, 0>, boost::numeric::ublas::vector<double>>, Kratos::UblasSpace<double, boost::numeric::ublas::matrix<double>, boost::numeric::ublas::vector<double>>, Kratos::LinearSolver<Kratos::UblasSpace<double, boost::numeric::ublas::compressed_matrix<double, boost::numeric::ublas::basic_row_major<>, 0>, boost::numeric::ublas::vector<double>>, Kratos::UblasSpace<double, boost::numeric::ublas::matrix<double>, boost::numeric::ublas::vector<double>>>>::Execute' requested here
    ~DistanceSmoothingProcess() override
    ^
/opt/homebrew/opt/llvm@14/bin/../include/c++/v1/__memory/unique_ptr.h:54:5: note: in instantiation of member function 'Kratos::DistanceSmoothingProcess<3, Kratos::UblasSpace<double, boost::numeric::ublas::compressed_matrix<double, boost::numeric::ublas::basic_row_major<>, 0>, boost::numeric::ublas::vector<double>>, Kratos::UblasSpace<double, boost::numeric::ublas::matrix<double>, boost::numeric::ublas::vector<double>>, Kratos::LinearSolver<Kratos::UblasSpace<double, boost::numeric::ublas::compressed_matrix<double, boost::numeric::ublas::basic_row_major<>, 0>, boost::numeric::ublas::vector<double>>, Kratos::UblasSpace<double, boost::numeric::ublas::matrix<double>, boost::numeric::ublas::vector<double>>>>::~DistanceSmoothingProcess' requested here
    delete __ptr;
    ^
/opt/homebrew/opt/llvm@14/bin/../include/c++/v1/__memory/unique_ptr.h:315:7: note: in instantiation of member function 'std::default_delete<Kratos::DistanceSmoothingProcess<3, Kratos::UblasSpace<double, boost::numeric::ublas::compressed_matrix<double, boost::numeric::ublas::basic_row_major<>, 0>, boost::numeric::ublas::vector<double>>, Kratos::UblasSpace<double, boost::numeric::ublas::matrix<double>, boost::numeric::ublas::vector<double>>, Kratos::LinearSolver<Kratos::UblasSpace<double, boost::numeric::ublas::compressed_matrix<double, boost::numeric::ublas::basic_row_major<>, 0>, boost::numeric::ublas::vector<double>>, Kratos::UblasSpace<double, boost::numeric::ublas::matrix<double>, boost::numeric::ublas::vector<double>>>>>::operator()' requested here
      __ptr_.second()(__tmp);
      ^
/opt/homebrew/opt/llvm@14/bin/../include/c++/v1/__memory/unique_ptr.h:269:19: note: in instantiation of member function 'std::unique_ptr<Kratos::DistanceSmoothingProcess<3, Kratos::UblasSpace<double, boost::numeric::ublas::compressed_matrix<double, boost::numeric::ublas::basic_row_major<>, 0>, boost::numeric::ublas::vector<double>>, Kratos::UblasSpace<double, boost::numeric::ublas::matrix<double>, boost::numeric::ublas::vector<double>>, Kratos::LinearSolver<Kratos::UblasSpace<double, boost::numeric::ublas::compressed_matrix<double, boost::numeric::ublas::basic_row_major<>, 0>, boost::numeric::ublas::vector<double>>, Kratos::UblasSpace<double, boost::numeric::ublas::matrix<double>, boost::numeric::ublas::vector<double>>>>>::reset' requested here
  ~unique_ptr() { reset(); }
                  ^
/opt/homebrew/opt/llvm@14/bin/../include/c++/v1/__memory/shared_ptr.h:458:25: note: in instantiation of member function 'std::unique_ptr<Kratos::DistanceSmoothingProcess<3, Kratos::UblasSpace<double, boost::numeric::ublas::compressed_matrix<double, boost::numeric::ublas::basic_row_major<>, 0>, boost::numeric::ublas::vector<double>>, Kratos::UblasSpace<double, boost::numeric::ublas::matrix<double>, boost::numeric::ublas::vector<double>>, Kratos::LinearSolver<Kratos::UblasSpace<double, boost::numeric::ublas::compressed_matrix<double, boost::numeric::ublas::basic_row_major<>, 0>, boost::numeric::ublas::vector<double>>, Kratos::UblasSpace<double, boost::numeric::ublas::matrix<double>, boost::numeric::ublas::vector<double>>>>>::~unique_ptr' requested here
        unique_ptr<_Yp> __hold(__p);
                        ^
/Users/wangxin/repos/Kratos/external_libraries/pybind11/stl/../pybind11.h:1835:61: note: in instantiation of function template specialization 'std::shared_ptr<Kratos::DistanceSmoothingProcess<3, Kratos::UblasSpace<double, boost::numeric::ublas::compressed_matrix<double, boost::numeric::ublas::basic_row_major<>, 0>, boost::numeric::ublas::vector<double>>, Kratos::UblasSpace<double, boost::numeric::ublas::matrix<double>, boost::numeric::ublas::vector<double>>, Kratos::LinearSolver<Kratos::UblasSpace<double, boost::numeric::ublas::compressed_matrix<double, boost::numeric::ublas::basic_row_major<>, 0>, boost::numeric::ublas::vector<double>>, Kratos::UblasSpace<double, boost::numeric::ublas::matrix<double>, boost::numeric::ublas::vector<double>>>>>::shared_ptr<Kratos::DistanceSmoothingProcess<3, Kratos::UblasSpace<double, boost::numeric::ublas::compressed_matrix<double, boost::numeric::ublas::basic_row_major<>, 0>, boost::numeric::ublas::vector<double>>, Kratos::UblasSpace<double, boost::numeric::ublas::matrix<double>, boost::numeric::ublas::vector<double>>, Kratos::LinearSolver<Kratos::UblasSpace<double, boost::numeric::ublas::compressed_matrix<double, boost::numeric::ublas::basic_row_major<>, 0>, boost::numeric::ublas::vector<double>>, Kratos::UblasSpace<double, boost::numeric::ublas::matrix<double>, boost::numeric::ublas::vector<double>>>>, void>' requested here
            new (std::addressof(v_h.holder<holder_type>())) holder_type(v_h.value_ptr<type>());
                                                            ^
/Users/wangxin/repos/Kratos/external_libraries/pybind11/stl/../pybind11.h:1850:9: note: in instantiation of member function 'pybind11::class_<Kratos::DistanceSmoothingProcess<3, Kratos::UblasSpace<double, boost::numeric::ublas::compressed_matrix<double, boost::numeric::ublas::basic_row_major<>, 0>, boost::numeric::ublas::vector<double>>, Kratos::UblasSpace<double, boost::numeric::ublas::matrix<double>, boost::numeric::ublas::vector<double>>, Kratos::LinearSolver<Kratos::UblasSpace<double, boost::numeric::ublas::compressed_matrix<double, boost::numeric::ublas::basic_row_major<>, 0>, boost::numeric::ublas::vector<double>>, Kratos::UblasSpace<double, boost::numeric::ublas::matrix<double>, boost::numeric::ublas::vector<double>>>>, std::shared_ptr<Kratos::DistanceSmoothingProcess<3, Kratos::UblasSpace<double, boost::numeric::ublas::compressed_matrix<double, boost::numeric::ublas::basic_row_major<>, 0>, boost::numeric::ublas::vector<double>>, Kratos::UblasSpace<double, boost::numeric::ublas::matrix<double>, boost::numeric::ublas::vector<double>>, Kratos::LinearSolver<Kratos::UblasSpace<double, boost::numeric::ublas::compressed_matrix<double, boost::numeric::ublas::basic_row_major<>, 0>, boost::numeric::ublas::vector<double>>, Kratos::UblasSpace<double, boost::numeric::ublas::matrix<double>, boost::numeric::ublas::vector<double>>>>>, Kratos::Process>::init_holder' requested here
        init_holder(inst, v_h, (const holder_type *) holder_ptr, v_h.value_ptr<type>());
        ^
/Users/wangxin/repos/Kratos/external_libraries/pybind11/stl/../pybind11.h:1524:32: note: in instantiation of member function 'pybind11::class_<Kratos::DistanceSmoothingProcess<3, Kratos::UblasSpace<double, boost::numeric::ublas::compressed_matrix<double, boost::numeric::ublas::basic_row_major<>, 0>, boost::numeric::ublas::vector<double>>, Kratos::UblasSpace<double, boost::numeric::ublas::matrix<double>, boost::numeric::ublas::vector<double>>, Kratos::LinearSolver<Kratos::UblasSpace<double, boost::numeric::ublas::compressed_matrix<double, boost::numeric::ublas::basic_row_major<>, 0>, boost::numeric::ublas::vector<double>>, Kratos::UblasSpace<double, boost::numeric::ublas::matrix<double>, boost::numeric::ublas::vector<double>>>>, std::shared_ptr<Kratos::DistanceSmoothingProcess<3, Kratos::UblasSpace<double, boost::numeric::ublas::compressed_matrix<double, boost::numeric::ublas::basic_row_major<>, 0>, boost::numeric::ublas::vector<double>>, Kratos::UblasSpace<double, boost::numeric::ublas::matrix<double>, boost::numeric::ublas::vector<double>>, Kratos::LinearSolver<Kratos::UblasSpace<double, boost::numeric::ublas::compressed_matrix<double, boost::numeric::ublas::basic_row_major<>, 0>, boost::numeric::ublas::vector<double>>, Kratos::UblasSpace<double, boost::numeric::ublas::matrix<double>, boost::numeric::ublas::vector<double>>>>>, Kratos::Process>::init_instance' requested here
        record.init_instance = init_instance;
                               ^
/Users/wangxin/repos/Kratos/applications/FluidDynamicsApplication/custom_python/add_custom_processes_to_python.cpp:166:5: note: in instantiation of function template specialization 'pybind11::class_<Kratos::DistanceSmoothingProcess<3, Kratos::UblasSpace<double, boost::numeric::ublas::compressed_matrix<double, boost::numeric::ublas::basic_row_major<>, 0>, boost::numeric::ublas::vector<double>>, Kratos::UblasSpace<double, boost::numeric::ublas::matrix<double>, boost::numeric::ublas::vector<double>>, Kratos::LinearSolver<Kratos::UblasSpace<double, boost::numeric::ublas::compressed_matrix<double, boost::numeric::ublas::basic_row_major<>, 0>, boost::numeric::ublas::vector<double>>, Kratos::UblasSpace<double, boost::numeric::ublas::matrix<double>, boost::numeric::ublas::vector<double>>>>, std::shared_ptr<Kratos::DistanceSmoothingProcess<3, Kratos::UblasSpace<double, boost::numeric::ublas::compressed_matrix<double, boost::numeric::ublas::basic_row_major<>, 0>, boost::numeric::ublas::vector<double>>, Kratos::UblasSpace<double, boost::numeric::ublas::matrix<double>, boost::numeric::ublas::vector<double>>, Kratos::LinearSolver<Kratos::UblasSpace<double, boost::numeric::ublas::compressed_matrix<double, boost::numeric::ublas::basic_row_major<>, 0>, boost::numeric::ublas::vector<double>>, Kratos::UblasSpace<double, boost::numeric::ublas::matrix<double>, boost::numeric::ublas::vector<double>>>>>, Kratos::Process>::class_<>' requested here
    py::class_<DistanceSmoothingProcess<3,SparseSpaceType,LocalSpaceType,LinearSolverType>, DistanceSmoothingProcess<3,SparseSpaceType,LocalSpaceType,LinearSolverType>::Pointer, Process>(m,"DistanceSmoothingProcess3D")
    ^
/opt/homebrew/opt/llvm@14/bin/../include/c++/v1/type_traits:3731:34: note: 'result_of<(lambda at /Users/wangxin/repos/Kratos/applications/FluidDynamicsApplication/custom_processes/distance_smoothing_process.h:203:13) (Kratos::GlobalPointer<Kratos::Node<3>> &)>' has been explicitly marked deprecated here
template <class _Callable> class _LIBCPP_DEPRECATED_IN_CXX17 result_of;
                                 ^
/opt/homebrew/opt/llvm@14/bin/../include/c++/v1/__config:1030:39: note: expanded from macro '_LIBCPP_DEPRECATED_IN_CXX17'
#  define _LIBCPP_DEPRECATED_IN_CXX17 _LIBCPP_DEPRECATED
                                      ^
/opt/homebrew/opt/llvm@14/bin/../include/c++/v1/__config:1007:48: note: expanded from macro '_LIBCPP_DEPRECATED'
#    define _LIBCPP_DEPRECATED __attribute__ ((deprecated))
                                               ^
In file included from /Users/wangxin/repos/Kratos/applications/FluidDynamicsApplication/custom_python/add_custom_processes_to_python.cpp:35:
In file included from /Users/wangxin/repos/Kratos/applications/FluidDynamicsApplication/custom_processes/distance_smoothing_process.h:23:
In file included from /Users/wangxin/repos/Kratos/kratos/processes/find_global_nodal_neighbours_process.h:28:
/Users/wangxin/repos/Kratos/kratos/utilities/pointer_communicator.h:67:27: warning: 'result_of<(lambda at /Users/wangxin/repos/Kratos/applications/FluidDynamicsApplication/custom_processes/distance_smoothing_process.h:210:13) (Kratos::GlobalPointer<Kratos::Node<3>> &)>' is deprecated [-Wdeprecated-declarations]
    typedef typename std::result_of< TFunctorType(GlobalPointer<TPointerDataType>&)>::type TSendType;
                          ^
/Users/wangxin/repos/Kratos/applications/FluidDynamicsApplication/custom_processes/distance_smoothing_process.h:209:43: note: in instantiation of template class 'Kratos::ResultsProxy<Kratos::Node<3>, (lambda at /Users/wangxin/repos/Kratos/applications/FluidDynamicsApplication/custom_processes/distance_smoothing_process.h:210:13)>' requested here
        auto contact_proxy = pointer_comm.Apply(
                                          ^
/Users/wangxin/repos/Kratos/applications/FluidDynamicsApplication/custom_processes/distance_smoothing_process.h:152:5: note: in instantiation of member function 'Kratos::DistanceSmoothingProcess<3, Kratos::UblasSpace<double, boost::numeric::ublas::compressed_matrix<double, boost::numeric::ublas::basic_row_major<>, 0>, boost::numeric::ublas::vector<double>>, Kratos::UblasSpace<double, boost::numeric::ublas::matrix<double>, boost::numeric::ublas::vector<double>>, Kratos::LinearSolver<Kratos::UblasSpace<double, boost::numeric::ublas::compressed_matrix<double, boost::numeric::ublas::basic_row_major<>, 0>, boost::numeric::ublas::vector<double>>, Kratos::UblasSpace<double, boost::numeric::ublas::matrix<double>, boost::numeric::ublas::vector<double>>>>::Execute' requested here
    ~DistanceSmoothingProcess() override
    ^
/opt/homebrew/opt/llvm@14/bin/../include/c++/v1/__memory/unique_ptr.h:54:5: note: in instantiation of member function 'Kratos::DistanceSmoothingProcess<3, Kratos::UblasSpace<double, boost::numeric::ublas::compressed_matrix<double, boost::numeric::ublas::basic_row_major<>, 0>, boost::numeric::ublas::vector<double>>, Kratos::UblasSpace<double, boost::numeric::ublas::matrix<double>, boost::numeric::ublas::vector<double>>, Kratos::LinearSolver<Kratos::UblasSpace<double, boost::numeric::ublas::compressed_matrix<double, boost::numeric::ublas::basic_row_major<>, 0>, boost::numeric::ublas::vector<double>>, Kratos::UblasSpace<double, boost::numeric::ublas::matrix<double>, boost::numeric::ublas::vector<double>>>>::~DistanceSmoothingProcess' requested here
    delete __ptr;
    ^
/opt/homebrew/opt/llvm@14/bin/../include/c++/v1/__memory/unique_ptr.h:315:7: note: in instantiation of member function 'std::default_delete<Kratos::DistanceSmoothingProcess<3, Kratos::UblasSpace<double, boost::numeric::ublas::compressed_matrix<double, boost::numeric::ublas::basic_row_major<>, 0>, boost::numeric::ublas::vector<double>>, Kratos::UblasSpace<double, boost::numeric::ublas::matrix<double>, boost::numeric::ublas::vector<double>>, Kratos::LinearSolver<Kratos::UblasSpace<double, boost::numeric::ublas::compressed_matrix<double, boost::numeric::ublas::basic_row_major<>, 0>, boost::numeric::ublas::vector<double>>, Kratos::UblasSpace<double, boost::numeric::ublas::matrix<double>, boost::numeric::ublas::vector<double>>>>>::operator()' requested here
      __ptr_.second()(__tmp);
      ^
/opt/homebrew/opt/llvm@14/bin/../include/c++/v1/__memory/unique_ptr.h:269:19: note: in instantiation of member function 'std::unique_ptr<Kratos::DistanceSmoothingProcess<3, Kratos::UblasSpace<double, boost::numeric::ublas::compressed_matrix<double, boost::numeric::ublas::basic_row_major<>, 0>, boost::numeric::ublas::vector<double>>, Kratos::UblasSpace<double, boost::numeric::ublas::matrix<double>, boost::numeric::ublas::vector<double>>, Kratos::LinearSolver<Kratos::UblasSpace<double, boost::numeric::ublas::compressed_matrix<double, boost::numeric::ublas::basic_row_major<>, 0>, boost::numeric::ublas::vector<double>>, Kratos::UblasSpace<double, boost::numeric::ublas::matrix<double>, boost::numeric::ublas::vector<double>>>>>::reset' requested here
  ~unique_ptr() { reset(); }
                  ^
/opt/homebrew/opt/llvm@14/bin/../include/c++/v1/__memory/shared_ptr.h:458:25: note: in instantiation of member function 'std::unique_ptr<Kratos::DistanceSmoothingProcess<3, Kratos::UblasSpace<double, boost::numeric::ublas::compressed_matrix<double, boost::numeric::ublas::basic_row_major<>, 0>, boost::numeric::ublas::vector<double>>, Kratos::UblasSpace<double, boost::numeric::ublas::matrix<double>, boost::numeric::ublas::vector<double>>, Kratos::LinearSolver<Kratos::UblasSpace<double, boost::numeric::ublas::compressed_matrix<double, boost::numeric::ublas::basic_row_major<>, 0>, boost::numeric::ublas::vector<double>>, Kratos::UblasSpace<double, boost::numeric::ublas::matrix<double>, boost::numeric::ublas::vector<double>>>>>::~unique_ptr' requested here
        unique_ptr<_Yp> __hold(__p);
                        ^
/Users/wangxin/repos/Kratos/external_libraries/pybind11/stl/../pybind11.h:1835:61: note: in instantiation of function template specialization 'std::shared_ptr<Kratos::DistanceSmoothingProcess<3, Kratos::UblasSpace<double, boost::numeric::ublas::compressed_matrix<double, boost::numeric::ublas::basic_row_major<>, 0>, boost::numeric::ublas::vector<double>>, Kratos::UblasSpace<double, boost::numeric::ublas::matrix<double>, boost::numeric::ublas::vector<double>>, Kratos::LinearSolver<Kratos::UblasSpace<double, boost::numeric::ublas::compressed_matrix<double, boost::numeric::ublas::basic_row_major<>, 0>, boost::numeric::ublas::vector<double>>, Kratos::UblasSpace<double, boost::numeric::ublas::matrix<double>, boost::numeric::ublas::vector<double>>>>>::shared_ptr<Kratos::DistanceSmoothingProcess<3, Kratos::UblasSpace<double, boost::numeric::ublas::compressed_matrix<double, boost::numeric::ublas::basic_row_major<>, 0>, boost::numeric::ublas::vector<double>>, Kratos::UblasSpace<double, boost::numeric::ublas::matrix<double>, boost::numeric::ublas::vector<double>>, Kratos::LinearSolver<Kratos::UblasSpace<double, boost::numeric::ublas::compressed_matrix<double, boost::numeric::ublas::basic_row_major<>, 0>, boost::numeric::ublas::vector<double>>, Kratos::UblasSpace<double, boost::numeric::ublas::matrix<double>, boost::numeric::ublas::vector<double>>>>, void>' requested here
            new (std::addressof(v_h.holder<holder_type>())) holder_type(v_h.value_ptr<type>());
                                                            ^
/Users/wangxin/repos/Kratos/external_libraries/pybind11/stl/../pybind11.h:1850:9: note: in instantiation of member function 'pybind11::class_<Kratos::DistanceSmoothingProcess<3, Kratos::UblasSpace<double, boost::numeric::ublas::compressed_matrix<double, boost::numeric::ublas::basic_row_major<>, 0>, boost::numeric::ublas::vector<double>>, Kratos::UblasSpace<double, boost::numeric::ublas::matrix<double>, boost::numeric::ublas::vector<double>>, Kratos::LinearSolver<Kratos::UblasSpace<double, boost::numeric::ublas::compressed_matrix<double, boost::numeric::ublas::basic_row_major<>, 0>, boost::numeric::ublas::vector<double>>, Kratos::UblasSpace<double, boost::numeric::ublas::matrix<double>, boost::numeric::ublas::vector<double>>>>, std::shared_ptr<Kratos::DistanceSmoothingProcess<3, Kratos::UblasSpace<double, boost::numeric::ublas::compressed_matrix<double, boost::numeric::ublas::basic_row_major<>, 0>, boost::numeric::ublas::vector<double>>, Kratos::UblasSpace<double, boost::numeric::ublas::matrix<double>, boost::numeric::ublas::vector<double>>, Kratos::LinearSolver<Kratos::UblasSpace<double, boost::numeric::ublas::compressed_matrix<double, boost::numeric::ublas::basic_row_major<>, 0>, boost::numeric::ublas::vector<double>>, Kratos::UblasSpace<double, boost::numeric::ublas::matrix<double>, boost::numeric::ublas::vector<double>>>>>, Kratos::Process>::init_holder' requested here
        init_holder(inst, v_h, (const holder_type *) holder_ptr, v_h.value_ptr<type>());
        ^
/Users/wangxin/repos/Kratos/external_libraries/pybind11/stl/../pybind11.h:1524:32: note: in instantiation of member function 'pybind11::class_<Kratos::DistanceSmoothingProcess<3, Kratos::UblasSpace<double, boost::numeric::ublas::compressed_matrix<double, boost::numeric::ublas::basic_row_major<>, 0>, boost::numeric::ublas::vector<double>>, Kratos::UblasSpace<double, boost::numeric::ublas::matrix<double>, boost::numeric::ublas::vector<double>>, Kratos::LinearSolver<Kratos::UblasSpace<double, boost::numeric::ublas::compressed_matrix<double, boost::numeric::ublas::basic_row_major<>, 0>, boost::numeric::ublas::vector<double>>, Kratos::UblasSpace<double, boost::numeric::ublas::matrix<double>, boost::numeric::ublas::vector<double>>>>, std::shared_ptr<Kratos::DistanceSmoothingProcess<3, Kratos::UblasSpace<double, boost::numeric::ublas::compressed_matrix<double, boost::numeric::ublas::basic_row_major<>, 0>, boost::numeric::ublas::vector<double>>, Kratos::UblasSpace<double, boost::numeric::ublas::matrix<double>, boost::numeric::ublas::vector<double>>, Kratos::LinearSolver<Kratos::UblasSpace<double, boost::numeric::ublas::compressed_matrix<double, boost::numeric::ublas::basic_row_major<>, 0>, boost::numeric::ublas::vector<double>>, Kratos::UblasSpace<double, boost::numeric::ublas::matrix<double>, boost::numeric::ublas::vector<double>>>>>, Kratos::Process>::init_instance' requested here
        record.init_instance = init_instance;
                               ^
/Users/wangxin/repos/Kratos/applications/FluidDynamicsApplication/custom_python/add_custom_processes_to_python.cpp:166:5: note: in instantiation of function template specialization 'pybind11::class_<Kratos::DistanceSmoothingProcess<3, Kratos::UblasSpace<double, boost::numeric::ublas::compressed_matrix<double, boost::numeric::ublas::basic_row_major<>, 0>, boost::numeric::ublas::vector<double>>, Kratos::UblasSpace<double, boost::numeric::ublas::matrix<double>, boost::numeric::ublas::vector<double>>, Kratos::LinearSolver<Kratos::UblasSpace<double, boost::numeric::ublas::compressed_matrix<double, boost::numeric::ublas::basic_row_major<>, 0>, boost::numeric::ublas::vector<double>>, Kratos::UblasSpace<double, boost::numeric::ublas::matrix<double>, boost::numeric::ublas::vector<double>>>>, std::shared_ptr<Kratos::DistanceSmoothingProcess<3, Kratos::UblasSpace<double, boost::numeric::ublas::compressed_matrix<double, boost::numeric::ublas::basic_row_major<>, 0>, boost::numeric::ublas::vector<double>>, Kratos::UblasSpace<double, boost::numeric::ublas::matrix<double>, boost::numeric::ublas::vector<double>>, Kratos::LinearSolver<Kratos::UblasSpace<double, boost::numeric::ublas::compressed_matrix<double, boost::numeric::ublas::basic_row_major<>, 0>, boost::numeric::ublas::vector<double>>, Kratos::UblasSpace<double, boost::numeric::ublas::matrix<double>, boost::numeric::ublas::vector<double>>>>>, Kratos::Process>::class_<>' requested here
    py::class_<DistanceSmoothingProcess<3,SparseSpaceType,LocalSpaceType,LinearSolverType>, DistanceSmoothingProcess<3,SparseSpaceType,LocalSpaceType,LinearSolverType>::Pointer, Process>(m,"DistanceSmoothingProcess3D")
    ^
/opt/homebrew/opt/llvm@14/bin/../include/c++/v1/type_traits:3731:34: note: 'result_of<(lambda at /Users/wangxin/repos/Kratos/applications/FluidDynamicsApplication/custom_processes/distance_smoothing_process.h:210:13) (Kratos::GlobalPointer<Kratos::Node<3>> &)>' has been explicitly marked deprecated here
template <class _Callable> class _LIBCPP_DEPRECATED_IN_CXX17 result_of;
                                 ^
/opt/homebrew/opt/llvm@14/bin/../include/c++/v1/__config:1030:39: note: expanded from macro '_LIBCPP_DEPRECATED_IN_CXX17'
#  define _LIBCPP_DEPRECATED_IN_CXX17 _LIBCPP_DEPRECATED
                                      ^
/opt/homebrew/opt/llvm@14/bin/../include/c++/v1/__config:1007:48: note: expanded from macro '_LIBCPP_DEPRECATED'
#    define _LIBCPP_DEPRECATED __attribute__ ((deprecated))
```

**runtime test output**

```
 |  /           |                  
 ' /   __| _` | __|  _ \   __|    
 . \  |   (   | |   (   |\__ \  
_|\_\_|  \__,_|\__|\___/ ____/
           Multi-Physics 9.2."0"-1acbb92cf6-Release-ARM64
           Compiled for Mac OS and Python3.10 with Clang-14.0
Compiled with threading support.
Maximum number of threads: 10.
```

--- a/kratos/includes/serializer.h
+++ b/kratos/includes/serializer.h
@@ -565,8 +565,18 @@ public:
 
         save("size", size);
 
+        // Workaround for the special case of std::vector<bool>.
+        // Long story short, the return type of std::vector<bool>::operator[]
+        // is not necessarily bool&, but up to the STL vendor's decision.
+        // Detailed explanation: https://github.com/KratosMultiphysics/Kratos/issues/10357#issuecomment-1274725614.
+        using SaveType = std::conditional_t<
+            std::is_same_v<typename std::decay<TDataType>::type, bool>,
+            bool,
+            const TDataType&
+        >;
+
         for(SizeType i = 0 ; i < size ; i++)
-            save("E", rObject[i]);
+            save("E", SaveType(rObject[i]));
 //    write(rObject);
     }
 

